WMF Framework/NavigationBar.swift

```diff
@@ -361,6 +361,20 @@ public class NavigationBar: SetupView, FakeProgressReceiving, FakeProgressDelega
             setNavigationBarPercentHidden(_navigationBarPercentHidden, underBarViewPercentHidden: _underBarViewPercentHidden, extendedViewPercentHidden: newValue, topSpacingPercentHidden: _topSpacingPercentHidden, animated: false)
         }
     }
+
+    private var shouldUnderBarIgnoreSafeArea: Bool = false {
+        didSet {
+            guard shouldUnderBarIgnoreSafeArea != oldValue else {
+                return
+            }
+
+            safeAreaLayoutGuide.leadingAnchor.constraint(equalTo: underBarView.leadingAnchor).isActive = !shouldUnderBarIgnoreSafeArea
+            safeAreaLayoutGuide.trailingAnchor.constraint(equalTo: underBarView.trailingAnchor).isActive = !shouldUnderBarIgnoreSafeArea
+
+            leadingAnchor.constraint(equalTo: underBarView.leadingAnchor).isActive = shouldUnderBarIgnoreSafeArea
+            trailingAnchor.constraint(equalTo: underBarView.trailingAnchor).isActive = shouldUnderBarIgnoreSafeArea
```

Comment:
`constraint(equalTo:` creates a new constraint and doesn't return an existing one, so new constraints will be created each time `shouldUnderBarIgnoreSafeArea` is set. Could these be created and stored in properties in `setup()`?
```
underBarRespectSafeAreaConstraints = [underBarViewLeadingConstraint, underBarViewTrailingConstraint]
underBarIgnoreSafeAreaConstraints = [...]
```
Then you could toggle `isActive` on those arrays here

Reply:
My bad - I had it in my head that identically created constraints were treated as if they were the same. Not sure where I got that from. Will update this shortly.